github/enterprise_manage_ghes.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2025 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+)
+
+// NodeQueryOptions specifies the optional parameters to the EnterpriseService
+// Node management APIS.
+type NodeQueryOptions struct {
+	// UUID filters issues based on the node UUID.
+	UUID *string `url:"uuid,omitempty"`
+
+	// ClusterRoles filters The cluster roles from the cluster configuration file.
+	ClusterRoles *string `url:"cluster_roles,omitempty"`
+}
+
+// ClusterStatus represents a response from the GetClusterStatus and GetReplicationStatus method.
+type ClusterStatus struct {
+	Status *string               `json:"status,omitempty"`
+	Nodes  []*ClusterStatusNodes `json:"nodes"`
+}
+
+// ClusterStatusNodes represents the status of a cluster node.
+type ClusterStatusNodes struct {
+	Hostname *string                       `json:"hostname,omitempty"`
+	Status   *string                       `json:"status,omitempty"`
+	Services []*ClusterStatusNodesServices `json:"services"`
+}
+
+// ClusterStatusNodesServices represents the status of a service running on a cluster node.
+type ClusterStatusNodesServices struct {
+	Status  *string `json:"status,omitempty"`
+	Name    *string `json:"name,omitempty"`
+	Details *string `json:"details,omitempty"`
+}
+
+// SystemRequirements represents a response from the GetCheckSystemRequirements method.
+type SystemRequirements struct {
+	Status *string                    `json:"status,omitempty"`
+	Nodes  []*SystemRequirementsNodes `json:"nodes"`
+}
+
+// SystemRequirementsNodes represents the status of a system node.
+type SystemRequirementsNodes struct {
+	Hostname    *string                               `json:"hostname,omitempty"`
+	Status      *string                               `json:"status,omitempty"`
+	RolesStatus []*SystemRequirementsNodesRolesStatus `json:"roles_status"`
+}
+
+// SystemRequirementsNodesRolesStatus represents the status of a role on a system node.
+type SystemRequirementsNodesRolesStatus struct {
+	Status *string `json:"status,omitempty"`
+	Role   *string `json:"role,omitempty"`
+}
+
+// NodeReleaseVersions represents a response from the GetReplicationStatus method.
+type NodeReleaseVersions struct {
+	Hostname *string          `json:"hostname,omitempty"`
+	Version  *ReleaseVersions `json:"version"`
+}
+
+// ReleaseVersions holds the release version information of the node.
+type ReleaseVersions struct {
+	Version   *string `json:"version,omitempty"`
+	Platform  *string `json:"platform,omitempty"`
+	BuildID   *string `json:"build_id,omitempty"`
+	BuildDate *string `json:"build_date,omitempty"`
```

Comment:
Should this be `*Timestamp` instead of `*string`?  I'll need to check the official docs, but don't have time right now.

Reply:
@gmlewis it's a string in the following format `2023-05-02` ([docs](https://docs.github.com/en/enterprise-server@3.15/rest/enterprise-admin/manage-ghes?apiVersion=2022-11-28#get-all-ghes-release-versions-for-all-nodes))